### PR TITLE
New and improved promise-based syntax examples and other edits

### PR DESCRIPTION
**Pull request by sidewayss**
The current page for decodeAudioData() is bad at explaining the newer promise-based syntax, and it is the newer, better way to go.  These edits update the page to reflect the newer reality of promises and async/await.
I have also created an updated example web page for webaudio-examples.  That PR is here: https://github.com/mdn/webaudio-examples/pull/104
I am a bit worried by a mistake I made in my last PR to this repo: I never created a separate branch, I committed to my main branch.  Now those commits are in this new dAD branch and I don't know if that screws up your merge.  Let me know if there is anything I should do about that and if I need to create a new PR as a result.

### CODE REVIEW

**Comment by sidewayss** [comment]
The lint failed because of a trailing space.  How do I figure out where to fix it?

**Comment by sidewayss** [comment]
The webaudio-examples PR is https://github.com/mdn/webaudio-examples/pull/104

**Comment by wbamberg** [comment]
Thanks for your PR, @sidewayss ! It looks fine. Don't worry about squashing commits, we do that on merge anyway. And although [your old commit](https://github.com/mdn/content/pull/26839/commits/dc3b41301ae1d7da173a1a8d3b4fbe3eff1b410f) is listed here: https://github.com/mdn/content/pull/26839/commits , it doesn't seem to show up in the diff - maybe something somewhere knows that the same commit is in main?

**Comment by sidewayss** [comment]
@wbamberg - great! How do we make sure that the example page PR is merged prior to this one?  Otherwise two links in this one won't work.

**Comment by wbamberg** [comment]
> @wbamberg - great! How do we make sure that the example page PR is merged prior to this one? Otherwise two links in this one won't work.

Good point. I'll review the example PR today.

**Comment by sidewayss** [comment]
@queengooborg Should I have run Prettier? That's what the Markdown linter seems to be saying.  Let me know if there's anything I need to do to proceed with your merge.

**Comment by queengooborg** [comment]
Hey @sidewayss!  We just merged an update to the linter workflow that enforces Prettier on most docs -- I went through all open PRs that don't have merge conflicts to update them and make sure that the linter passes the Prettier check as well.  If you could run Prettier on the affected file, that would be great!

**Comment by sidewayss** [comment]
@queengooborg Give me a few minutes here to install Node.js and the npm CLI.  I'll commit the changes here when I'm done (maybe tomorrow as it's getting late here).  I assume you'll be notified when that happens.

**Comment by sidewayss** [comment]
@queengooborg - OK, I've got Prettier installed now, but in reading the bit of docs I've read it looks like I need to use specific options or create a config file with those options.  Are there specific options I should set?  I'm on Windows right now, if that makes a difference.  Or should I just run `>npx prettier index.md` and overwrite my local file with the default options?

**Comment by queengooborg** [comment]
> in reading the bit of docs I've read it looks like I need to use specific options or create a config file with those options.

Those docs are written for getting started on integrating Prettier into projects, rather than using it on projects that already have Prettier!  All of the options and configuration has already been set in this project, so all you will need to do is run `npm run fix:md` in this repository (or `npx prettier --write [file]`) and it will fix all of the issues found!

**Comment by sidewayss** [comment]
Well I've already run prettier and directed stdout to a new file, which I attached here.  If that file is "good to go", then I'll rename it to overwrite index.md and commit it.  I'll do that now.
I'm curious about the `npm run fix:md` command.  I can just run that from this decodeaudiodata directory and it's the same as `prettier --write`?

**Comment by sidewayss** [comment]
A related question that I don't know where to ask:
The docs page for the `hsl()` color function has an interactive example that is coded this way in the index.md file:
`{{EmbedInteractiveExample("pages/css/function-hsl.html")}}`
The newer color functions, like `lab()` don't have interactive examples and I was thinking it would be easy to add them, but I can't find the `pages` sub-directory.  Where can I find these examples to view the html files and create a new ones for these new functions?

**Comment by sidewayss** [comment]
I think I'm going to have to close this PR and create a new one because of a lack of verified commits.  Plus this entire fork of mine is tainted by my initial commits to the main branch.  I plan to copy this one file offline, then delete this repo and fork from origin all over again.  I still have to update the XHR example in this docs page and make sure the two examples align well.  I'll probably do it and be ready with a new PR tomorrow.

**Comment by wbamberg** [comment]
> A related question that I don't know where to ask: The docs page for the `hsl()` color function has an interactive example that is coded this way in the index.md file: `{{EmbedInteractiveExample("pages/css/function-hsl.html")}}` The newer color functions, like `lab()` don't have interactive examples and I was thinking it would be easy to add them, but I can't find the `pages` sub-directory. Where can I find these examples to view the html files and create a new ones for these new functions?

Sorry I missed this q before. Interactive examples live in yet another repo: https://github.com/mdn/interactive-examples.

**Comment by sidewayss** [comment]
No problem, it wasn't urgent.  Thanks for the info.

**Comment by sidewayss** [comment]
Wow, those two paths are far from being the same.
The embed path: `{{EmbedInteractiveExample("pages/css/function-hsl.html")}}`
vs
The path relative to the interactive-examples project: `live-examples/css-examples/color/function-hsl.html`
I would never have found it on my own.  Thanks!

File: files/en-us/web/api/baseaudiocontext/decodeaudiodata/index.md [hunk 1]
```diff
@@ -22,21 +22,20 @@ data.
 ## Syntax
 
 ```js-nolint
-// Older callback syntax:
+// Promise-based syntax:
+promise = decodeAudioData(arrayBuffer)
```
Comment (Josh-Cena) [inline files/en-us/web/api/baseaudiocontext/decodeaudiodata/index.md:26]:
```suggestion
decodeAudioData(arrayBuffer)
```

**Comment by sidewayss** [comment]
@Josh-Cena - Why? decodeAudioData returns a promise which is resolved later.  It doesn't return nothing.  You are undoing a change I made intentionally.  What did I do wrong?  Please explain.

**Comment by sidewayss** [comment]
Thanks for responding.  I realize that uniformity is important, but this is an unusual function.  I think I can find a way to emphasize the differences in the two ways of using this function while maintaining this particular uniformity.  If I keep the comments and remove the return value from the "code" in the suggestion that I made, it will likely make everyone happy.
I was waiting until tomorrow morning or until I got a response to this issue before continuing with changes here.  Tomorrow for sure I'll close this PR and open a new one with a clean repo/fork and a verified commit.  If there's anything still out of line you or someone else can correct it and I'll accept the changes without question.

So yes, please deter me from opening any cans of worms.  I don't want to rock any boats here.  I just want to make sure this exceptional function's exceptional qualities are clearly explained.  Are there any other web api functions like this that have two entirely different ways of using them?  From what I've seen a new function is usually introduced to implement the new behavior, not a new set of features for the existing function.  But I'm not familiar with the whole api.

**Comment by sidewayss** [comment]
This PR has been replaced by: #27185.

**Pull request closed without merging**